lib/granite/action/preconditions.rb

```diff
@@ -105,8 +106,15 @@ def decline_with(*args, **kwargs)
 
       private
 
-      def run_preconditions!
-        _preconditions.execute! self
+      def run_preconditions!(cache_result: false)
+        if cache_result
+          if @preconditions_run.nil?
+            _preconditions.execute!(self)
+            @preconditions_run = true
+          end
+        else
+          _preconditions.execute!(self) unless @preconditions_run
+        end
```

Comment:
IIUC we can change it to:
```suggestion
        _preconditions.execute!(self) if @preconditions_run.nil? || !cache_result

        @preconditions_run = true if cache_result
```

Reply:
I've added a spec and your version doesn't fix the bug, I'm afraid

Reply:
Sorry, I misunderstood, the second call is `run_preconditions!(cache_result: false)` but we sill want to use the cached results, I think this should do:

```suggestion
       _preconditions.execute!(self) if @preconditions_run.nil?                                                                                                                  
                                                                                                                                                                                    
       @preconditions_run = true if cache_result
```

So IIUC:

```ruby
# Case 1 with cache
run_preconditions!(cache_result: true) # Run preconditions and cache the result
run_preconditions!(cache_result: false) # DO not run preconditions and use the cache 

# Case 2 without cache
run_preconditions!(cache_result: false) # Run preconditions
run_preconditions!(cache_result: false) # Run preconditions
```

Maybe would be good to cover these two cases in `spec/lib/granite/action/preconditions_spec.rb` , WDYT?